src/Microsoft.Identity.Client/Internal/IdToken.cs

```diff
@@ -73,22 +73,33 @@ internal class IdToken
 
         public static IdToken Parse(string idToken)
         {
+            if (string.IsNullOrEmpty(idToken))
+            {
+                return null;
+            }
+
             IdToken idTokenBody = null;
-            if (!string.IsNullOrWhiteSpace(idToken))
+            string[] idTokenSegments = idToken.Split(new[] {'.'});
+
+            if (idTokenSegments.Length < 2)
             {
-                string[] idTokenSegments = idToken.Split(new[] {'.'});
+                throw new MsalClientException(MsalClientException.InvalidJwtError, "ID Token must contain at least 2 parts.");
```

Comment:
kk. Followed up offline, we'll still treat jwt error as client side error since the communication with server succeeds, the failures comes when client side is trying to read the jwt, which should be an exception on the client side.